crates/shielded_token/src/masp.rs

```diff
@@ -118,21 +116,16 @@ struct MaspTargetTransferData {
 pub struct MaspDataLog {
     pub source: Option<TransferSource>,
     pub token: Address,
-    pub amount: token::DenominatedAmount,
+    pub amount: token::Amount,
```

Comment:
This thing is causing issues here #3961. It seems like without the denomination we interpret the amounts wrong leading the huge values